dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/internal/PersistentExecutorIntrospector.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.persistent.internal;
+
+import java.io.PrintWriter;
+import java.security.AccessController;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.websphere.concurrent.persistent.PersistentExecutor;
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.kernel.service.util.SecureAction;
+import com.ibm.wsspi.logging.Introspector;
+
+/**
+ * Introspector for persistent executors.
+ */
+@Component(configurationPolicy = ConfigurationPolicy.IGNORE)
+public class PersistentExecutorIntrospector implements Introspector {
+    @Override
+    @Trivial
+    public String getIntrospectorName() {
+        System.out.println("getIntrospectorName");
+        return "PersistentExecutorIntrospector";
+    }
+
+    @Override
+    @Trivial
+    public String getIntrospectorDescription() {
+        System.out.println("getIntrospectorDescription");
+        return "Persistent timers/tasks diagnostics";
+    }
+
+    @Override
+    public void introspect(PrintWriter out) throws Exception {
+        SecureAction priv = AccessController.doPrivileged(SecureAction.get());
+        BundleContext bundleContext = priv.getBundleContext(FrameworkUtil.getBundle(getClass()));
+
+        for (ServiceReference<PersistentExecutor> ref : priv.getServiceReferences(bundleContext, PersistentExecutor.class, "(!(com.ibm.wsspi.resource.ResourceFactory=true))")) {
+            System.out.println("Found service reference: " + ref);
```

Comment:
Same question as above.

Reply:
good catch, those were just for debugging locally. I'll remove them.